### PR TITLE
chore(CSharp): added merge sort

### PR DESCRIPTION
**Pull request by Waqar-107**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
added merge sort implementation with unit test

**Other information:**

### CODE REVIEW

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @Waqar-107,
Please add reference on this [file](https://github.com/Waqar-107/DSA/blob/merge-sort/docs/en/Sorting/Merge-Sort.md)

File: algorithms/CSharp/README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ For running the `.cs` file please using [.Net Finddle](https://dotnetfiddle.net/
 2. [Insertion Sort](src/Sorts/insertion-sort.cs)
 3. [Selection Sort](src/Sorts/selection-sort.cs)
 4. [Counting Sort](src/Sorts/counting-sort.cs)
+5. [Merge Sort](src/Sorts/merge-sort.cs)
```
Comment (Waqar-107) [inline algorithms/CSharp/README.md:10]: @ming-tsai by reference, didn't you mean this? I can see it's already in the PR. Let me know if I got it wrong

Comment (ming-tsai) [reply algorithms/CSharp/README.md:10]: In this case is adding to documentations folder, due to this is algorithm is documented

Comment (Waqar-107) [reply algorithms/CSharp/README.md:10]: @ming-tsai got it. Adding it asap

Comment (Waqar-107) [reply algorithms/CSharp/README.md:10]: @ming-tsai done

**Review by ming-tsai — APPROVED** [review]

**Pull request merged**